components/match2/wikis/tft/match_group_input_custom.lua

```diff
@@ -0,0 +1,181 @@
+---
+-- @Liquipedia
+-- wiki=tft
+-- page=Module:MatchGroup/Input/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Array = require('Module:Array')
+local FnUtil = require('Module:FnUtil')
+local Lua = require('Module:Lua')
+local Operator = require('Module:Operator')
+local Table = require('Module:Table')
+local Variables = require('Module:Variables')
+
+local MatchGroupInputUtil = Lua.import('Module:MatchGroup/Input/Util')
+local Streams = Lua.import('Module:Links/Stream')
+
+local DEFAULT_BESTOF = 3
+local DEFAULT_MODE = 'team'
+
+local CustomMatchGroupInput = {}
+local MatchFunctions = {}
+local MapFunctions = {}
+
+CustomMatchGroupInput.processMap = FnUtil.identity
+
+---@param match table
+---@param options table?
+---@return table
+function CustomMatchGroupInput.processMatch(match, options)
+	local finishedInput = match.finished --[[@as string?]]
+	local winnerInput = match.winner --[[@as string?]]
+
+	Table.mergeInto(match, MatchGroupInputUtil.readDate(match.date))
+
+	local opponents = Array.mapIndexes(function(opponentIndex)
+		return MatchGroupInputUtil.readOpponent(match, opponentIndex)
+	end)
+
+	local games = MatchFunctions.extractMaps(match, opponents)
+
+	local autoScoreFunction = MatchGroupInputUtil.canUseAutoScore(match, games)
+		and MatchFunctions.calculateMatchScore(games)
+		or nil
+
+	Array.forEach(opponents, function(opponent, opponentIndex)
+		opponent.score, opponent.status = MatchGroupInputUtil.computeOpponentScore({
+			walkover = match.walkover,
+			winner = match.winner,
+			opponentIndex = opponentIndex,
+			score = opponent.score,
+		}, autoScoreFunction)
+	end)
+
+	match.finished = MatchGroupInputUtil.matchIsFinished(match, opponents)
+
+	if match.finished then
+		match.resulttype = MatchGroupInputUtil.getResultType(winnerInput, finishedInput, opponents)
+		match.walkover = MatchGroupInputUtil.getWalkover(match.resulttype, opponents)
+		match.winner = MatchGroupInputUtil.getWinner(match.resulttype, winnerInput, opponents)
+		MatchGroupInputUtil.setPlacement(opponents, match.winner, 1, 2, match.resulttype)
+	end
+
+	MatchFunctions.getTournamentVars(match)
+
+	match.stream = Streams.processStreams(match)
+	match.links = MatchFunctions.getLinks(match)
+
+	match.games = games
+	match.opponents = opponents
+
+	match.extradata = MatchFunctions.getExtraData(match)
+
+	return match
+end
+
+---@param match table
+---@return table
+function MatchFunctions.getTournamentVars(match)
+	match.mode = Variables.varDefault('tournament_mode', DEFAULT_MODE)
+	return MatchGroupInputUtil.getCommonTournamentVars(match)
+end
+
+---@param match table
+---@return table
+function MatchFunctions.getLinks(match)
+	return {}
+end
+
+---@param match table
+---@return table
+function MatchFunctions.getExtraData(match)
+	return {
+		mvp = MatchGroupInputUtil.readMvp(match),
+		mapveto = MatchGroupInputUtil.getMapVeto(match),
+		casters = MatchGroupInputUtil.readCasters(match),
```

Comment:
Are all these 3 actually used on the wiki?

Reply:
no idea

@Hesketh2 do you know if they need it?

Reply:
![image](https://github.com/user-attachments/assets/723d8b7f-7840-4948-b0c5-8a32aea76e36)
Based on EWC looks like only MVP/Caster is, but not Veto

Reply:
alright will kick the veto start then